### PR TITLE
Fix Blazor E2E tests and remove obsolete BindMethods code

### PR DESCRIPTION
**Pull request by rynowak**
See commits for details

### CODE REVIEW

**Review by pranavkm — APPROVED** [review]

**Comment by SteveSandersonMS** [comment]
The debug version of the .js files does exist on my machine after building. Did something change with this recently? Or is there some other reason why it needs to be hardcoded as Release now?

**Comment by rynowak** [comment]
> The debug version of the .js files does exist on my machine after building. Did something change with this recently? Or is there some other reason why it needs to be hardcoded as Release now?

I have no idea what changed. I have been unable to run the client-side E2E tests since I got back from Spain.

The `Debug` folder is created when you build but it's not checked in. I don't know what the current state is supposed to be because honestly, this all seems unacceptable.

**Comment by rynowak** [comment]
I'm removing the hardcoding to unblock this. Now you'll get an error if you try to build a one of these projects without having the appropriate JS files present. I think this is an improvement to the current state because it now fails noisily instead of silently.

**Comment by SteveSandersonMS** [comment]
> I don't know what the current state is supposed to be because honestly, this all seems unacceptable.

Totally agree - there are multiple problems with the build currently. For me, not running the E2E tests in CI is the worst of them.

> Now you'll get an error if you try to build a one of these projects without having the appropriate JS files present. I think this is an improvement to the current state because it now fails noisily instead of silently.

Good, I agree that's an improvement!

File: src/Components/test/testassets/TestServer/Startup.cs [hunk 1]
```diff
@@ -92,6 +84,14 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
...
                 });
             });
+
+            app.UseRouting();
+
+            app.UseEndpoints(endpoints =>
+            {
+                endpoints.MapControllers();
+                endpoints.MapRazorPages();
+            });
```
Comment (javiercn) [inline src/Components/test/testassets/TestServer/Startup.cs]: I believe I know the answer, but why move this to the bottom?

Comment (rynowak) [reply src/Components/test/testassets/TestServer/Startup.cs]: I'm not aware of a real problem here in the way we're using this website, but the old code allowed routing to match (and execute) for URLs that are inside `/prerendered`. That's definitely not intended. I found this while debugging something and decided to clean it up.

**Comment by rynowak** [comment]
> The Debug js files aren't checked in so we'll have to find out if this breaks people's workflow or not, but I still see it as an improvement.

Note that this breaks `build --all` but it was broken already (invalid build output due to silent failure).

**Pull request merged**